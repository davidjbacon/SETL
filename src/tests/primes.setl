PROGRAM primes;
  PRINTA (STDERR, 'Please enter n, the ceiling on prime numbers:');
  READ (x);
  n := FLOOR x;
  s := {2..n} - {i*j : i IN {2..FLOOR SQRT n}, j IN {i..n DIV 2}};
  t := {p IN {2..n} | FORALL i IN {2..FLOOR SQRT p} | p MOD i /= 0};
  ASSERT s = t;
  PRINT (t);
END PROGRAM primes;
