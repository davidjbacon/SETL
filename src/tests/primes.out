{257 769 2 3 5 773 7 263 521 11 523 13 269 271 17 19 787 277 23 281 283 29 541 797 31 547 37 293 41 809 43 811 557 47 307 563 53 821 311 823 313 569 59 571 827 61 317 829 577 67 71 839 73 331 587 79 337 593 83 853 599 89 601 857 347 859 349 607 863 97 353 101 613 103 359 617 107 619 109 877 367 113 881 883 373 631 887 379 127 383 641 131 643 389 647 137 139 907 397 653 911 401 659 149 661 151 919 409 157 673 929 163 419 421 677 167 937 683 173 941 431 433 179 691 947 181 439 953 443 701 191 193 449 197 709 199 967 457 971 461 463 719 977 211 467 727 983 733 223 479 991 227 739 229 997 487 743 233 491 239 751 241 499 757 503 761 251 509}
